src/server/routes.js

```diff
@@ -58,18 +60,22 @@ app.post("/inventory/product", function (req, res) {
   const document = req.body;
 
   if (
-    !document.product_id ||
-    !document.style_id ||
-    !document.size ||
-    !document.quantity
+    document.product_id === undefined ||
+    document.style_id === undefined ||
+    document.size === undefined ||
+    document.quantity === undefined
   ) {
     res.sendStatus(400);
     return;
   }
 
   controller
     .write(document)
-    .then(function (result) {
+    .then(async function (result) {
+      const key = `/inventory/${document.product_id}/${document.style_id}`;
+
+      // Remove existing key to prevent cache from sending in an old version after a post request
+      await cache.remove(key);
```

Comment:
**Annotation**: Rather than updating the stored item, remove the item in cache entirely so a new updated version can be stored later on. Although this may hinder the runtime for retrieving the specified item after the POST method, it's less complex than updating the entire body stored, meaning less potential errors. Additionally, the body stored in the cache is an array of products, and modifying one item in the array may not be linear time.